pkg/spire/spire_mock.go

```diff
@@ -0,0 +1,273 @@
+*/
+
+package spire
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SpireMockClient is a client used for mocking the this package for unit testing
+// other tekton components that use the spire entrypointer or controller client.
+//
+// The SpireMockClient implements both SpireControllerApiClient and SpireEntrypointerApiClient
+// and in addition to that provides the helper functions to define and query internal state.
+type SpireMockClient struct {
+	// Entries is a dictionary of entries that mock the SPIRE server datastore (for function Sign only)
+	Entries map[string]bool
+
+	// SignIdentities represents the list of identities to use to sign (providing context of a caller to Sign)
+	// when Sign is called, the identity is dequeued from the slice. A signature will only be provided if the
+	// corresponding entry is in Entries. This only takes effect if SignOverride is nil.
+	SignIdentities []string
+
+	// VerifyAlwaysReturns defines whether to always verify successfully or to always fail verification if non-nil.
+	// This only take effect on Verify functions:
+	// - VerifyStatusInternalAnnotationOverride
+	// - VerifyTaskRunResultsOverride
+	VerifyAlwaysReturns *bool
+
+	// VerifyStatusInternalAnnotationOverride contains the function to overwrite a call to VerifyStatusInternalAnnotation
+	VerifyStatusInternalAnnotationOverride func(ctx context.Context, tr *v1beta1.TaskRun, logger *zap.SugaredLogger) error
+
+	// VerifyTaskRunResultsOverride contains the function to overwrite a call to VerifyTaskRunResults
+	VerifyTaskRunResultsOverride func(ctx context.Context, prs []v1beta1.PipelineResourceResult, tr *v1beta1.TaskRun) error
+
+	// AppendStatusInternalAnnotationOverride  contains the function to overwrite a call to AppendStatusInternalAnnotation
+	AppendStatusInternalAnnotationOverride func(ctx context.Context, tr *v1beta1.TaskRun) error
+
+	// CheckSpireVerifiedFlagOverride contains the function to overwrite a call to CheckSpireVerifiedFlag
+	CheckSpireVerifiedFlagOverride func(tr *v1beta1.TaskRun) bool
+
+	// SignOverride contains the function to overwrite a call to Sign
+	SignOverride func(ctx context.Context, results []v1beta1.PipelineResourceResult) ([]v1beta1.PipelineResourceResult, error)
+}
+
+const (
+	controllerSvid = "CONTROLLER_SVID_DATA"
+)
+
+func (_ *SpireMockClient) mockSign(content, signedBy string) string {
+	return fmt.Sprintf("signed-by-%s:%x", signedBy, sha256.Sum256([]byte(content)))
+}
+
+func (sc *SpireMockClient) mockVerify(content, sig, signedBy string) bool {
+	return sig == sc.mockSign(content, signedBy)
+}
+
+func (_ *SpireMockClient) GetIdentity(tr *v1beta1.TaskRun) string {
+	return fmt.Sprintf("/ns/%v/taskrun/%v", tr.Namespace, tr.Name)
+}
+
+func (sc *SpireMockClient) AppendStatusInternalAnnotation(ctx context.Context, tr *v1beta1.TaskRun) error {
```

Comment:
missing a check for `AppendStatusInternalAnnotationOverride`

Reply:
fixed